aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx

```diff
@@ -104,17 +116,32 @@ function ExtensionModal({ isOpen, handleClose, selectedExtension }: ExtensionMod
     const { loading, mutate } = useExecuteExtension();
     const [inputParams, setInputParams] = useState<Array<actionInExtensionsType>>()
     const { setSelectedFile } = useContext(FileSelectStateContext)
+    const [executeExtensionEnabled, setExecuteExtensionEnabled] = useState(false);
 
-    //initiate input params when selected extension changes
     useEffect(() => {
         setInputParams(selectedAction?.parameters.map(
             item => (item.dataType === ExtensionParameterDataTypes.EXPERIMENT
                 ? { name: item.name, value: experimentIdentifier }
-                : { name: item.name, value: item.defaultValue }
+                : { name: item.name, value: item.defaultValue ?? "" }
             ))
         )
     }, [selectedExtension, selectedAction])
 
+    useEffect(() => {
+        let enableExecutionUpdate: boolean = true;
+        enableExecutionUpdate = !inputParams?.some(param => {
+            const paramElement = selectedAction?.parameters?.find(inputParam => inputParam.name === param.name);
+            return (
+                (paramElement?.dataType === ExtensionParameterDataTypes.STR && param.value === "") ||
+                (paramElement?.dataType === ExtensionParameterDataTypes.INT && param.value === "") ||
+                (paramElement?.dataType === ExtensionParameterDataTypes.FLOAT && param.value === "") ||
+                (paramElement?.dataType === ExtensionParameterDataTypes.SELECT && param.value === "") ||
+                (paramElement?.dataType === ExtensionParameterDataTypes.TEXTAREA && param.value === "")
+            );
+        });
+        setExecuteExtensionEnabled(enableExecutionUpdate);
+    }, [inputParams, selectedAction])
+
```

Comment:
we can replace it all with this variable
```suggestion
    const isExtensionExexutable = inputParams?.every(param => param.value)
```